lib/unifex/specs.ex

```diff
@@ -0,0 +1,153 @@
+defmodule Unifex.Specs do
+  @moduledoc """
+  Module exporting macros that can be used to define Unifex specs.
+
+  Example of such specs is provided below:
+
+      module Membrane.Element.Mad.Decoder.Native
+
+      spec create() :: {:ok :: label, state}
+
+      spec decode_frame(payload, offset :: int, state) ::
+             {:ok :: label, {payload, bytes_used :: long, sample_rate :: long, channels :: int}}
+             | {:error :: label, :buflen :: label}
+             | {:error :: label, :malformed :: label}
+             | {:error :: label, {:recoverable :: label, bytes_to_skip :: int}}
+
+      dirty :cpu, decode_frame: 3
+
+      sends {:example_msg ::label, number :: int}
+
+  According to this specification, module `Membrane.Element.Mad.Decoder.Native` should contain 2 functions: `create/0`
+  and `decode_frame/3` (which is a cpu-bound dirty NIF). The module should use `Unifex.Loader` to provide access to
+  these functions. What is more, messages of the form `{:example_msg, integer}` can be sent from the native code to
+  erlang processes.
+
+  The generated boilerplate would require implementation of the following functions:
+
+      UNIFEX_TERM create(UnifexEnv* env);
+      UNIFEX_TERM decode_frame(UnifexEnv* env, UnifexPayload * payload, int offset, State* state);
+
+  Also the following functions that should be called to return results will be generated:
+
+      UNIFEX_TERM create_result_ok(UnifexEnv* env, State* state);
+      UNIFEX_TERM decode_frame_result_ok(UnifexEnv* env, UnifexPayload * payload,
+                                         long bytes_used, long sample_rate, int channels);
+      UNIFEX_TERM decode_frame_result_error_buflen(UnifexEnv* env);
+      UNIFEX_TERM decode_frame_result_error_malformed(UnifexEnv* env);
+      UNIFEX_TERM decode_frame_result_error_recoverable(UnifexEnv* env, int bytes_to_skip);
+
+  See docs for appropriate macros for more details.
+  """
+
+  @doc """
+  Macro used for defining module that is to export native functions to Elixir world.
```

Comment:
That sentence is quite hard to understand.
* `Macro used for defining module` -> `Defines module` sounds better and isn't that complicated.
* `that is to export native functions to Elixir world.` Again - overly complicated. I would rephrase it to something like `that exports native functions ..`